### DISCUSSION TITLE
When I try to use --data to replicate a previous transaction made in GUI the token value is incorrect

### DISCUSSION THREAD

**Discussion by 621625** [gochain/web3#232] [Q&A]
Hello,

I'm MB. A very nice and polite guy.

When I try to use --data to replicate a previous transaction made in GUI I get insufficient input amount error

Example:

Successful transaction made in GUI that I try to replicate:

https://polygonscan.com/tx/0xadae7f41832e9a506dab9558817fcb4ed44315b57145821a3c1abcd050ffcc3a

My input command:

web3 contract call --address 0x1b02da8cb0d097eb8d57a175b88c7d8b47997506 --data 0x7ff36ab500000000000000000000000000000000000000000000000011188dffee43975c000000000000000
000000000000000000000000000000000000000000
0000080000000000000000000000000bf5a722ace4
f8d8feaf802933e3ed90e090483740000000000000
000000000000000000000000000000000000000000
061678085000000000000000000000000000000000
000000000000000000000000000000200000000000
00000000000000d500b1d8e8ef31e21c99d1db9a64
44d3adf12700000000000000000000000008f3cf7a
d23cd3cadbd9735aff958023239c6a063

Resulting transaction (this transaction has 0 Matic value and fails):

https://polygonscan.com/tx/0xbd6b3a6c3f985c02c65aa869639b053a860b26e8f2ce181915b8ddaf9732eb14

How can I replicate the transaction including the correct value? Just using the same --data binary input is not enough.

Update: I tried putting --amount 1 before --address but then it uses "1 wei" instead of 1 Matic.

**Comment by 621625** [comment]
Update: I tried putting --amount 1 before --address but then it uses "1 wei" instead of 1 Matic.

**Comment by treeder** [comment]
`value` is separate from data, so you'll still need to pass a value. Add 18 zeroes to it to make 1 MATIC.

**Comment by 621625** [comment]
That solved the issue with the value.

https://polygonscan.com/tx/0xac3ad77e416f4612778e7f2659f31f972df9209eaf21e548dab986242231ab5b

Now I get an error message saying "expired!" even though the data is exactly the same as in the initial transaction. Any idea why?

-------- Original Message --------

**Reply by treeder** [reply]
No idea, probably a question for polygon or sushiswap or whatever it is you're using.
